arangod/RocksDBEngine/RocksDBEdgeIndex.cpp

```diff
@@ -282,57 +334,65 @@ class RocksDBEdgeIndexLookupIterator final : public IndexIterator {
       _lastKey = _keysIterator.value();
       TRI_ASSERT(_lastKey.isString());
       std::string_view fromTo = _lastKey.stringView();
+      std::string_view cacheKey;
 
       bool needRocksLookup = true;
       if (_cache) {
-        // Try to read from cache
-        auto finding =
-            _cache->find(fromTo.data(), static_cast<uint32_t>(fromTo.size()));
-        if (finding.found()) {
-          incrCacheHits();
-          needRocksLookup = false;
-          // We got sth. in the cache
-          VPackSlice cachedData(finding.value()->value());
-          TRI_ASSERT(cachedData.isArray());
-          if (cachedData.length() / 2 < limit) {
-            // Directly return it, no need to copy
-            _builderIterator = VPackArrayIterator(cachedData);
-            while (_builderIterator.valid()) {
-              TRI_ASSERT(_builderIterator.value().isNumber());
-              LocalDocumentId docId{
-                  _builderIterator.value().getNumericValue<uint64_t>()};
-
-              _builderIterator.next();
-
-              TRI_ASSERT(_builderIterator.valid());
-              TRI_ASSERT(_builderIterator.value().isString());
-              std::forward<F>(cb)(docId, _builderIterator.value());
-
-              _builderIterator.next();
-              limit--;
+        // try to read from cache
+
+        // build actual cache lookup key. this is either the value of fromTo,
+        // or a suffix of it in case fromTo starts with the cached collection
+        // name, and an empty string_view if the lookup value is invalid.
+        cacheKey = _index->buildCompressedCacheKey(cacheKeyCollection, fromTo);
+        TRI_ASSERT(cacheKey.empty() || cacheKey == fromTo ||
```

Comment:
NIT
Maybe move assert under !cacheKey.empty()?

In such case first part of disjunction can be removed